tests/UI/tools/stepIdentifiersChecker.ts

```diff
@@ -0,0 +1,190 @@
+import fs from 'fs';
+
+type JsonReportTest = {
+  context?: string,
+  file: string,
+  pending: boolean,
+  fullTitle: string
+}
+type JsonReportSuite = {
+  file: string,
+  tests: Array<JsonReportTest>
+  suites: Array<JsonReportSuite>
+}
+type JsonReportResult = {
+  suites?: Array<JsonReportSuite>
+}
+type JsonReport = {
+  results: Array<JsonReportResult>
+}
+
+// Get mochawesome report
+const reportPath: string = process.env.REPORT_PATH || './mochawesome-report/mochawesome.json';
+const jsonFile: string = fs.readFileSync(reportPath).toString();
+
+/**
+ * Get all tests from json file
+ * @param jsonFile
+ * @returns {[]}
+ */
+const getAllTests = (jsonFile: string): JsonReportTest[] => {
+  // Parse json report
+  const jsonReport: JsonReport = JSON.parse(jsonFile);
+  const parentSuites = jsonReport.results[0].suites;
+
+  let allTests: JsonReportTest[] = [];
+
+  // eslint-disable-next-line no-restricted-syntax
+  if (parentSuites !== undefined) {
+    parentSuites.forEach((suite:JsonReportSuite): void => {
+      allTests = allTests.concat(getTestsFromSuite(suite));
+    });
+  }
+
+  return allTests;
+};
+
+/**
+ * Get tests from suite and nested suites
+ * @param suite
+ * @returns {JsonReportTest[]}
+ */
+const getTestsFromSuite = (suite: JsonReportSuite): JsonReportTest[] => {
+  const {file} = suite;
+  let tests = suite.tests || [];
+  tests.forEach((test: JsonReportTest) => {
+    const testReturned = test;
+    testReturned.file = file;
+
+    return testReturned;
+  });
+
+  const nestedSuites: JsonReportSuite[] = suite.suites;
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const nestedSuite of nestedSuites) {
+    tests = tests.concat(getTestsFromSuite(nestedSuite));
+  }
+
+  return tests;
+};
+
+/**
+ * Check for undefined context
+ * @param jsonFile
+ */
+const checkUndefined = (jsonFile: string): boolean => {
+  const allTests = getAllTests(jsonFile);
+  const undefinedContextsSteps: string[] = allTests
+    .filter((test: JsonReportTest) => (test.context === undefined || !test.context) && !test.pending)
+    .map((test: JsonReportTest) => test.fullTitle.trim());
```

Comment:
could be reduced to only one loop by using `Array.reduce`, increasing performances